replica/distributed_loader.cc

```diff
@@ -391,14 +391,10 @@ future<> table_populator::populate_subdir(sstables::sstable_state state, allow_o
         return sst->get_origin() != sstables::repair_origin;
     };
 
-    // FIXME: Bypass reshape as it is not tablet aware, so it could mix sstables from different tablets together.
-    //  Refs: https://github.com/scylladb/scylladb/issues/16966.
-    if (!_global_table->uses_tablets()) {
-        co_await distributed_loader::reshape(directory, _db, sstables::reshape_mode::relaxed, _ks, _cf, [this, state](shard_id shard) {
-            auto gen = _global_table->calculate_generation_for_new_table();
-            return make_sstable(*_global_table, state, gen, _highest_version);
-        }, eligible_for_reshape_on_boot);
-    }
+    co_await distributed_loader::reshape(directory, _db, sstables::reshape_mode::relaxed, _ks, _cf, [this, state](shard_id shard) {
+        auto gen = _global_table->calculate_generation_for_new_table();
+        return make_sstable(*_global_table, state, gen, _highest_version);
+    }, eligible_for_reshape_on_boot);
 
```

Comment:
can you please write a python test for it (in core itself, not a dtest)? 

1) single cpu node with tablet enabled table containing 2-4 tablets
2) disable auto compaction
3) generate more than max_threshold (defaults to 32) sstables with help of 'nodetool flush', each sstable having at least a few keys to span multiple tablets
4) restart node, wait for reshape completed message in the log (there's a log watcher function).

Reply:
Done. Added a new testcase to topology_custom/test_tablet.py.